### ISSUE TITLE
Can we update same field twice in a function...

### ISSUE THREAD

**Issue by KamranAslamBhatti** [tfsaggregator/aggregator-cli#88]
Hi,
I have one single function which is mapped to update event, now in that function I have two separate code sections which both updates to same field i.e. TargetDate. When this function runs it gives me following error:
**An item with same key has already been added.**

I also try to create two different function on update event and these both function updating the same field on some conditions but this also did not work, out of two only one function runs and other function gives 404 Not found error.

Here is my code which I am using in a single function to update same field and getting **An item with same key already been added** error. 
Is there a way to update same field twice in a same function?

```csharp
//1) Logic 1
//**********

var childrens = self.Children;
if (childrens != null)
{
 foreach (var child in self.Children)
 {
  var childDate =child["Microsoft.VSTS.Scheduling.TargetDate"]?.ToString() ?? DateTime.Now.ToString("u");
  child["Microsoft.VSTS.Scheduling.TargetDate"]=DateTime.Parse(childDate.ToString()).AddDays(2);
 }
}
```

```csharp
//2) Logic 2
************
var allWorkItemLinks = self.RelationLinks;
if (allWorkItemLinks != null)
{
foreach(var successorLink in allWorkItemLinks.Where(link => string.Equals("System.LinkTypes.Dependency-Forward", link.Rel)))
{
   var id = int.Parse(successorLink.Url.Split('/').Last());
   var successor = store.GetWorkItem(id);
   var successorDate =DateTime.Parse(successor["Microsoft.VSTS.Scheduling.TargetDate"]?.ToString() ?? DateTime.Now.ToString("u"));
   successor["Microsoft.VSTS.Scheduling.TargetDate"]=DateTime.Parse(successorDate.ToString()).AddDays(2);
}
}
```

Thanks,
Kamran

**Comment by giuliov** [comment]
Sounds like a bug, but need to investigate.
I guess you have a workitem linked both as a child and as a successor.

**Comment by BobSilent** [comment]
Do you have a little bit more log context information about the error ** An item with same key already been added**

I have an idea, i think it is returned from Server not (directly) from our Code. Sure the error is at our side.

**Comment by BobSilent** [comment]
I tested a little bit around, there are two issues in code

1. An item with same key already been addedAn item with same key already been added
Occurs when a field not yet assigned (without value yet) in work item will be assigned twice

2. You cannot update the same field will occurs from server, if you update a field twice, which already had a value in received work item (e.g. title)

I am over it to create a fix

**Comment by giuliov** [comment]
Fixed in [0.9.14](https://github.com/tfsaggregator/aggregator-cli/releases/tag/v0.9.14)
